### PR TITLE
ensure bare AckNxt requesting >1 is supported

### PR DESCRIPTION
**Pull request by ripienaar**
Signed-off-by: R.I.Pienaar <rip@devco.net>

/cc @nats-io/core

### CODE REVIEW

**Comment by ripienaar** [comment]
looks to me the intention was that this is now supported but it didnt work

File: server/consumer.go [hunk 1]
```diff
@@ -1070,11 +1072,20 @@ func nextReqFromMsg(msg []byte) (time.Time, int, bool, error) {
...
 		return cr.Expires, cr.Batch, cr.NoWait, nil
 	}
-	// Naked batch size here for backward compatibility.
+
 	bs := 1
-	if n, err := strconv.Atoi(req); err == nil {
-		bs = n
+	// Naked batch size here for backward compatibility.
+	switch {
+	case bytes.HasPrefix(msg, AckNext):
```
Comment (derekcollison) [inline server/consumer.go:1079]: I think use `req` not `msg` here.

File: server/consumer.go [hunk 2]
```diff
@@ -1070,11 +1072,20 @@ func nextReqFromMsg(msg []byte) (time.Time, int, bool, error) {
...
+		if n, _ := fmt.Sscanf(string(msg), ackNextCnt, &bs); n == 0 {
```
Comment (derekcollison) [inline server/consumer.go:1080]: Same here. Not for prefix of course but possible trailing spaces.

File: test/jetstream_test.go [hunk 3]
```diff
@@ -82,6 +83,7 @@ func RunBasicJetStreamServer() *server.Server {
 	opts := DefaultTestOptions
 	opts.Port = -1
 	opts.JetStream = true
+	opts.NoLog = true
```
Comment (derekcollison) [inline test/jetstream_test.go:86]: This should be part of DefaultTestOptions no?

File: test/jetstream_test.go [hunk 4]
```diff
@@ -3013,10 +3015,92 @@ func TestJetStreamConsumerAckAck(t *testing.T) {
 	testAck(server.AckAck)
 	testAck(server.AckNak)
 	testAck(server.AckProgress)
-	testAck(server.AckNext)
 	testAck(server.AckTerm)
 }
 
+func TestJetStreamNack(t *testing.T) {
```
Comment (derekcollison) [inline test/jetstream_test.go:3021]: Test name wrong?

**Review by derekcollison — CHANGES_REQUESTED** [review]

File: test/jetstream_test.go (hunk 3 above)
Comment (ripienaar) [reply test/jetstream_test.go:86]: yeah not sure how that crept in there, removing

File: test/jetstream_test.go [hunk 5]
```diff
@@ -83,7 +83,7 @@ func RunBasicJetStreamServer() *server.Server {
 	opts := DefaultTestOptions
 	opts.Port = -1
 	opts.JetStream = true
-	opts.NoLog = true
+	opts.NoLog = false
```
Comment (derekcollison) [inline test/jetstream_test.go:86]: Do you want the test to print log statements? This should be set to true.

**Review by derekcollison — CHANGES_REQUESTED** [review]

File: test/jetstream_test.go (hunk 3 above)
Comment (kozlovic) [reply test/jetstream_test.go]: Actually, for websocket/mqtt tests, I do set this intentionally to `NoLog = false` and set a dummy logger with debug and trace enabled:
```
    opts.NoLog = false
    l := &DummyLogger{}
    s.SetLogger(l, true, true)
```
The reason? This exercises all code that has debug/trace, making sure that we don't have situations where we may be tracing something that causes a dereference of a nil pointer. As an added bonus, it helps with code coverage ;-)

File: test/jetstream_test.go (hunk 5 above)
Comment (kozlovic) [reply test/jetstream_test.go:86]: Also see my comment about testing debug/trace statements with the use of a dummy logger. Note, we may not have a DummyLogger in the test/ package, but that could be easily copied from server tests.

Comment (ripienaar) [reply test/jetstream_test.go]: restored as it was before, i think this is some line of code i put in on some other day while debugging and just didnt notice :) gone now.

File: test/jetstream_test.go (hunk 3 above)
Comment (derekcollison) [reply test/jetstream_test.go]: Might be a good idea.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by ripienaar** [comment]
merging after travis passed

**Pull request merged**